packages/rum/src/domain/segmentCollection/deflateWorkerSingleton.ts

```diff
@@ -1,32 +1,97 @@
 import { addErrorToMonitoringBatch, display, includes, monitor } from '@datadog/browser-core'
 import { createDeflateWorker, DeflateWorker } from './deflateWorker'
 
-let workerSingleton: DeflateWorker
-
-export function getDeflateWorkerSingleton() {
-  if (!workerSingleton) {
-    try {
-      workerSingleton = createDeflateWorker()
-    } catch (error) {
-      display.error('Session Replay recording failed to start: an error occurred while creating the Worker:', error)
-      if (includes(error.message, 'Content Security Policy')) {
-        display.error(
-          'Please make sure CSP is correctly configured ' +
-            'https://docs.datadoghq.com/real_user_monitoring/faq/content_security_policy'
-        )
-      } else {
-        addErrorToMonitoringBatch(error)
-      }
-      return
+const enum DeflateWorkerSingletonStatus {
+  Nil,
+  Loading,
+  Error,
+  Ready,
+}
```

Comment:
It could be interesting to add some explanations about what we want to solve with these states